sdk/src/executor/process/types.rs

```diff
@@ -73,123 +74,127 @@ impl GadgetProcess {
     /// Will loop and wait for output from the process, returns early if no output is received
     /// for a default timeout period of 1 second.
     pub(crate) async fn read_until_default_timeout(&mut self) -> ProcessOutput {
         self.read_until_timeout(DEFAULT_READ_TIMEOUT).await
     }
 
-    /// Continually reads output from this GadgetProcess, eventually returning a ProcessOutput.
+    /// Continually reads output from this [GadgetProcess], eventually returning a [ProcessOutput].
     /// Will loop and wait for output to contain the specified substring.
     pub async fn read_until_receiving_string(&mut self, substring: String) -> ProcessOutput {
         let mut messages = Vec::new();
         if let Some(stream) = &mut self.stream {
             // Read lines until we receive the desired substring
             loop {
-                let read_result = stream.next_line().await;
+                let read_result = stream.recv().await;
                 match read_result {
                     Ok(output) => {
-                        match output {
-                            None => {
-                                // Stream is completed - process is finished
-                                println!(
-                                    "{} : STREAM COMPLETED - ENDING",
-                                    self.process_name.to_string_lossy()
-                                );
-                                return ProcessOutput::Exhausted(messages);
-                            }
-                            Some(streamed_output) => {
-                                // We received output from child process
-                                // TODO: Log
-                                println!(
-                                    "{} : MESSAGE LOG : {}",
-                                    self.process_name.to_string_lossy(),
-                                    streamed_output.clone()
-                                );
-                                messages.push(streamed_output.clone());
-                                self.output.push(streamed_output.clone());
-                                if streamed_output.contains(&substring) {
-                                    // We should now return with the output
-                                    return ProcessOutput::Output(messages);
-                                }
+                        let inbound_message = output;
+                        if inbound_message.is_empty() {
+                            // Stream is completed - process is finished
+                            info!(
+                                "{} : STREAM COMPLETED - ENDING",
+                                self.process_name.to_string_lossy()
+                            );
+                            return ProcessOutput::Exhausted(messages);
+                        } else {
+                            // We received output from child process
+                            info!(
```

Comment:
Should these be info or debug logs? Can we do a cleanup of all the logging or at least let me know for sure everything is needed? It feels not like full mess, but it seems like some logs are in all caps which could lead me to believe they were for debugging?